experiments/simple_train_step.py

```diff
@@ -0,0 +1,31 @@
+from itertools import chain
+import torch
+
+
+def simple_train_step(optimizer,criterion,device,model,split_val):
+        
+    def train_step(x, y):
+        model.to(device)
```

Comment:
this does not need to happen every step -> extract this to a higher layer in the logic